diff --git a/pyproject.toml b/pyproject.toml
index d03036e7..2ee8a169 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -1,3 +1,36 @@
+# Per:
+# https://stackoverflow.com/questions/73600082/how-to-reference-a-requirements-txt-in-the-pyproject-toml-of-a-setuptools-projec
+[project]
+name = "stable_diffusion_webui"
+# authors = [
+#   "John Smith <johnsmith@example.org>",
+#   #"automatic111 <automatic1111@some-valid-email.com>"
+# ]
+description = "A Web UI for Stable Diffusion."
+version = "v1.8.0"
+
+dynamic = ["dependencies"]
+
+[[project.authors]]
+name = "Equinor"
+email = "fg_sib-scout@equinor.com"
+
+[tool.setuptools]
+py-modules = []
+
+[tool.setuptools.dynamic]
+dependencies = {file = ["requirements.txt"]}
+# [tool.poetry]
+# This keeps Poetry from complaining about pyproject.py being in the root
+# directory.
+# package-mode = false
+
+[build-system]
+requires = ["setuptools", "setuptools-scm"]
+build-backend = "setuptools.build_meta"
+# requires = ["poetry-core>=1.0.0"]
+# build-backend = "poetry.core.masonry.api"
+
 [tool.ruff]
 
 target-version = "py39"
